js/ui/form/ui.form.js

```diff
@@ -1259,15 +1264,26 @@ var Form = Widget.inherit({
         switch(rootOptionName) {
             case "items":
                 var itemPath = this._getItemPath(nameParts),
-                    instance,
-                    item = this.option(itemPath);
+                    item = this.option(itemPath),
+                    instance = this._instanceStorage.findByItem(item),
+                    fullName = args.fullName;
 
-                if(args.fullName.search("editorOptions") !== -1) {
-                    instance = this.getEditor(item.dataField) || this.getEditor(item.name);
-                    instance && instance.option(item.editorOptions);
+                if(instance) {
+                    if(fullName.search("buttonOptions") !== -1) {
+                        instance.option(item.buttonOptions);
+                        break;
+                    } else if(instance && fullName.search("editorOptions") !== -1) {
```

Comment:
You've already checked that `instance` exists at [L1267](https://github.com/DevExpress/DevExtreme/pull/5871/files#diff-f0dcc74cd8849c7f64781a7e90b0dd21R1267)